main.tf

```diff
@@ -50,6 +50,7 @@ resource "aws_lb_target_group" "this" {
   health_check {
     healthy_threshold   = var.health_check.healthy_threshold
     interval            = var.health_check.interval
+    path                = var.health_check.path
```

Comment:
`path` isn't valid for `health_check` when target group protocol is `TCP`.

Reply:
I like the idea of codifying that logic but if I forget that TCP can't have a health check I'd rather have it throw an error and remind me, not silently fix it for me. It's not like it gives me the option of not specifying `health_check.path`. Because it's an object I'll have to pass something in.

Reply:
`null` should exclude with most providers. You could just default null, or you could go the extra mile and implement my suggestion which protects from user error.